deploy/gcp/main.tf

```diff
@@ -308,62 +308,77 @@ resource "null_resource" "setup-env" {
   depends_on = [
     google_container_cluster.cluster,
     null_resource.get-credentials,
+    var.tidb_operator_registry,
+    var.tidb_operator_version,
   ]
 
   provisioner "local-exec" {
     working_dir = path.module
+    interpreter = ["bash", "-c"]
 
     command = <<EOS
-kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $(gcloud config get-value account)
-kubectl create serviceaccount --namespace kube-system tiller
+set -euo pipefail
+
+if ! kubectl get clusterrolebinding cluster-admin-binding 2>/dev/null; then
+  kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $(gcloud config get-value account)
+fi
+
+if ! kubectl get serviceaccount -n kube-system tiller 2>/dev/null ; then
+  kubectl create serviceaccount --namespace kube-system tiller
+fi
+
 kubectl apply -f manifests/crd.yaml
 kubectl apply -k manifests/local-ssd
 kubectl apply -f manifests/gke/persistent-disk.yaml
 kubectl apply -f manifests/tiller-rbac.yaml
+
 helm init --service-account tiller --upgrade --wait
 until helm ls; do
   echo "Wait until tiller is ready"
 done
-helm install --namespace tidb-admin --name tidb-operator ${path.module}/charts/tidb-operator
+helm upgrade --install tidb-operator --namespace tidb-admin ${path.module}/charts/tidb-operator --set operatorImage=${var.tidb_operator_registry}/tidb-operator:${var.tidb_operator_version}
 EOS
 
 
-environment = {
-KUBECONFIG = local.kubeconfig
-}
-}
+    environment = {
+      KUBECONFIG = local.kubeconfig
+    }
+  }
 }
 
 resource "null_resource" "deploy-tidb-cluster" {
-depends_on = [
-null_resource.setup-env,
-local_file.tidb-cluster-values,
-google_container_node_pool.pd_pool,
-google_container_node_pool.tikv_pool,
-google_container_node_pool.tidb_pool,
-]
-
-triggers = {
-values = data.template_file.tidb_cluster_values.rendered
-}
+  depends_on = [
+    null_resource.setup-env,
+    local_file.tidb-cluster-values,
+    google_container_node_pool.pd_pool,
+    google_container_node_pool.tikv_pool,
+    google_container_node_pool.tidb_pool,
+  ]
 
-provisioner "local-exec" {
+  triggers = {
+    values = data.template_file.tidb_cluster_values.rendered
+  }
+
+  provisioner "local-exec" {
+    interpreter = ["bash", "-c"]
 command = <<EOS
```

Comment:
Use `EOT` instead of `EOS` to make Terraform 0.12 format this correctly.

Reply:
any word will work here.